tests/test_odcs_requester.py

```diff
@@ -0,0 +1,83 @@
+"""test_odcs_requester.py - test odcs_requester"""
+from typing import Callable
+
+# pylint: disable=redefined-outer-name
+from unittest.mock import MagicMock, create_autospec
+
+import pytest
+from odcs.client.odcs import ODCS, ComposeSourceTag  # type: ignore
+
+from generate_compose.odcs_configurations_generator import (
+    ODCSComposeConfig,
+    ODCSComposesConfigs,
+)
+from generate_compose.odcs_requester import ODCSRequester, ODCSRequestReferences
+
+
+@pytest.fixture()
+def compose_url() -> str:
+    """Example compose-url, as close to the one expected"""
+    return "http://download.eng.bos.redhat.com/odcs/prod/odcs-222222"
+
+
+@pytest.fixture()
+def create_odcs_mock(compose_url: str) -> Callable[[int], MagicMock]:
+    """Create an ODCS mock with specific results for the compose method"""
+
+    def _mock_odcs_compose(num_of_composes: int) -> MagicMock:
+        """Monkey-patched ODCS compose"""
+        mock: MagicMock = create_autospec(ODCS)
+        mock.request_compose.side_effect = [
+            {"result_repofile": f"{compose_url}", "id": i}
+            for i in range(1, num_of_composes + 1)
+        ]
+        return mock
+
+    return _mock_odcs_compose
+
+
+@pytest.fixture()
+def compose_source_tag() -> ComposeSourceTag:
+    """Example ComposeSource of type 'tag'"""
+    return ComposeSourceTag(tag="tag")
+
+
+@pytest.mark.parametrize(
+    ("composes_configs, num_of_composes"),
+    [
+        pytest.param(
+            ODCSComposesConfigs([ODCSComposeConfig(spec=ComposeSourceTag(tag="tag"))]),
+            1,
+            id="single compose, tag source, no additional arguments",
+        ),
+        pytest.param(
+            ODCSComposesConfigs(
+                [
+                    ODCSComposeConfig(spec=ComposeSourceTag(tag="tag")),
+                    ODCSComposeConfig(spec=ComposeSourceTag(tag="tag")),
+                ]
+            ),
+            2,
```

Comment:
This can be `len(composes_configs.configs)` added to line 74

Reply:
That's what I started with, I don't mind using it. I thought the parametrize will be more explicit.

Reply:
Changed